src/audio/pipeline/pipeline-schedule.c

```diff
@@ -51,6 +51,9 @@ static enum task_state pipeline_task_cmd(struct pipeline *p,
 		case COMP_TRIGGER_PAUSE:
 			return p->trigger.aborted ? SOF_TASK_STATE_RUNNING :
 				SOF_TASK_STATE_COMPLETED;
+		case COMP_TRIGGER_PRE_START:
+		case COMP_TRIGGER_PRE_RELEASE:
+			p->status = COMP_STATE_ACTIVE;
```

Comment:
@plbossart this addresses a problem with demux pipelines, not with mixers. And by the way those pipelines seem to violate our current convention, that with such forking topologies each part should be a separate pipeline, i.e. with a "2-tooth fork" we should have at least 3 pipelines.
We currently have a lot of complexity, that handles different streaming states on different "teeth" of a mixer / mux / demux fork. And I have a dream (sorry, a plan) to simplify them along the lines of https://github.com/lyakh/sof/commit/760b0fb9e458d13e9deb353f5c34193195158e99 - some time when I have time... Then everything will be better, of course it will ;-)

Reply:
Thanks for pointing out this topology issue @lyakh. Something we need to look into before the transition to topology 2.0. @bardliao FYI.

And yes, we do need a plan to handle 'complex' graphs, the DPCM story tells us we have to look into this soon...

Reply:
This is on @lyakh plan for after his vacation - we need a simplification of pipeline logic together with some ref counting (missing today, we carry corner case logic today in generic pipe code)